### PR TITLE
Make image FIPS compatible

### PR DESCRIPTION
**Pull request by weinimo**
Closes: [OSPRH-3107](https://issues.redhat.com//browse/OSPRH-3107)

### CODE REVIEW

**Comment by weinimo** [comment]
/hold
testing

**Comment by lhh** [comment]
`CGO_ENABLED=1` is a baseline requirement to use FIPS-validated openssl encryption libraries.

File: Dockerfile
Comment (abays) [inline Dockerfile:1]: It looks like we're also adding `USER root` at line 21 as part of the pattern: https://github.com/openstack-k8s-operators/keystone-operator/pull/364/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557

Comment (weinimo) [reply Dockerfile:1]: Thanks for the hint!

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**